src/shared-components/button/style.ts

```diff
@@ -7,21 +7,37 @@ import { textColorsAssociatedWithColors } from './constants';
 
 import { ButtonTypeWithAction } from '.';
 
-const primaryStyles = (buttonColor: ThemeColors, theme: ThemeType) => `
-  background-color: ${buttonColor};
-  border-color: ${buttonColor};
-  color: ${theme.COLORS.white};
-  fill: ${theme.COLORS.white};
+const primaryStyles = (buttonColor: ThemeColors, theme: ThemeType) => {
+  let backgroundColor = buttonColor;
```

Comment:
Made my [previous comment](https://github.com/curology/radiance-ui/pull/651/files#r544659164) before seeing more bespoke logic. I think we should move logic like this (that breaks the normal 1:1 theme mapping of styling) into a separate file, or if we use a special convention for how to name these exceptions so that they're easily findable across the application. (I think the separate file is preferable, because it's more easily identifiable, ideally with a big JSDoc comment explaining the intention of the file). 

I imagine that there will be more exceptions as we catch them in review so we might as well start with this PR.